hasher-matcher-actioner/terraform/submit_events/main.tf

```diff
@@ -0,0 +1,185 @@
+      SUBMISSIONS_QUEUE_URL = var.submissions_queue.url
+    }
+  }
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventFunction"
+    }
+  )
+}
+
+resource "aws_cloudwatch_log_group" "submit_event_handler" {
+  name              = "/aws/lambda/${aws_lambda_function.submit_event_handler.function_name}"
+  retention_in_days = var.log_retention_in_days
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventLambdaLogGroup"
+    }
+  )
+}
+
+resource "aws_iam_role" "submit_event_handler" {
+  name_prefix        = "${var.prefix}_submit_event_handler"
+  assume_role_policy = data.aws_iam_policy_document.lambda_assume_role.json
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "SubmitEventLambdaRole"
+    }
+  )
+}
+
+data "aws_iam_policy_document" "submit_event_handler" {
+  statement {
+    effect    = "Allow"
+    actions   = ["dynamodb:GetItem", "dynamodb:Query", "dynamodb:Scan", "dynamodb:PutItem", "dynamodb:UpdateItem"]
+    resources = ["${var.datastore.arn}*"]
+  }
+
+  statement {
+    effect = "Allow"
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+      "logs:DescribeLogStreams"
+    ]
+    resources = ["${aws_cloudwatch_log_group.submit_event_handler.arn}:*"]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["cloudwatch:GetMetricStatistics"]
+    resources = ["*"]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["sqs:GetQueueAttributes", "sqs:ReceiveMessage", "sqs:DeleteMessage"]
+    resources = [aws_sqs_queue.submit_event_queue.arn]
+  }
+
+  statement {
+    effect    = "Allow"
+    actions   = ["sqs:SendMessage"]
+    resources = [var.submissions_queue.arn]
+  }
+  dynamic "statement" {
+    for_each = var.partner_image_buckets
+
+    content {
+      actions   = ["s3:GetObject"]
+      effect    = "Allow"
+      resources = ["${lookup(statement.value, "arn", null)}/*"]
+    }
+  }
+  # statement {
+  #   effect = "Allow"
+  #   actions = [
+  #     "s3:GetObject",
+  #   ]
+  #   resources = concat(
+  #     [for partner_bucket in var.partner_image_buckets : "${partner_bucket.arn}/*"]
+  #   )
+  # }
+}
+
+resource "aws_iam_policy" "submit_event_handler" {
+  name_prefix = "${var.prefix}_submit_event_handler_role_policy"
+  description = "Permissions for Root API Lambda"
+  policy      = data.aws_iam_policy_document.submit_event_handler.json
+}
+
+resource "aws_iam_role_policy_attachment" "submit_event_handler" {
+  role       = aws_iam_role.submit_event_handler.name
+  policy_arn = aws_iam_policy.submit_event_handler.arn
+}
+
+resource "aws_lambda_event_source_mapping" "submit_to_event_handler" {
+  event_source_arn = aws_sqs_queue.submit_event_queue.arn
+  function_name    = aws_lambda_function.submit_event_handler.arn
+
+  # Evidently, once set, batch-size and max-batching-window must always
+  # be provided. Else terraform warns.
+  batch_size                         = 1
```

Comment:
This would mean a lambda invocation per submission. Which is terribly costly. Is there a reason for not wanting to batch?

I'm assuming the actual runtime of the lambda is insignificant and memory bound (only IO is dropping message onto submissions queue). So we'll be paying a lot for lambda init.

Reply:
Was testing one at a time. Will change to 10